### PR TITLE
[Carina Ganea] functors_task_1

### PR DESCRIPTION
**Pull request by carina-eliza-ganea**
I was running the code from WSL Ubuntu 22 in VS Code, the build with make succeeded, but there were dependencies errors when adding executables for the unit tests, so I could not test the code.

### CODE REVIEW

**Comment by smay1613** [comment]
Please write me on email about the errors you have

File: functors_task_1/student/include/bar_serving.h [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #pragma once
 #include <array>
+using namespace std;
```
Comment (smay1613) [inline functors_task_1/student/include/bar_serving.h]: That's a bad practice 
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf6-use-using-namespace-directives-for-transition-for-foundation-libraries-such-as-std-or-within-a-local-scope-only

File: functors_task_1/student/include/impl.h [hunk 2]
```diff
@@ -1,13 +1,22 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <vector>
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: You're not need it yet

File: functors_task_1/student/include/impl.h [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+    int next = static_cast<int>(BeerBrand::HoeGaarden);
+    
+    BeerBrand operator()(){
+        if(next == static_cast<int>(BeerBrand::Max)){
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: please store initial value somewhere

File: functors_task_1/student/include/impl.h [hunk 4]
```diff
@@ -16,16 +25,18 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(BeerBrand beer)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h]: _Mark function params const by default, as modifying them is a bad practice_

**Comment by carina-eliza-ganea** [comment]
I managed to get passed the errors, they were due to a misunderstanding of the CMakeLists on my part. Everything seems to run smoothly now, including the tests.

File: functors_task_1/student/include/impl.h [hunk 5]
```diff
@@ -16,16 +26,18 @@ struct BeerOrganizer
...
 {
+    return beer == BeerBrand::Corona || beer == BeerBrand::HoeGaarden;
 }
 
 /**
  * @todo Implement lambda beer country equality comparator
  * @return true if beer county is the same, false otherwise
  */
-auto sameCountry = [](/**???*/)
+auto sameCountry = [](BeerBrand brand, BeerBrand beer)
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:38]: _also const_

File: functors_task_1/student/include/impl.h [hunk 6]
```diff
@@ -40,9 +52,30 @@ struct MixingPolicy
...
+                    case NonAlcoholDrink::GrapefruitJuice:
+                        return Cocktail::Greyhount;
+                        break;
+                    default:
+                        return Cocktail::Oops;
+                        break;
+                }
+                break;
+            case AlcoholDrink::Whiskey:
+                return (juice == NonAlcoholDrink::SevenUp)?Cocktail::SevenPlusSeven:Cocktail::Oops;
```
Comment (smay1613) [inline functors_task_1/student/include/impl.h:72]: _reconsider fomatting here_

**Pull request closed without merging**